### ISSUE TITLE
xcode-locator-bin warning in 0.3.2rc3

### ISSUE THREAD

**Issue by pcj** [bazelbuild/bazel#1875]
After I upgraded to 0.3.2rc3 I now routinely get the following warning:

<code>
WARNING: /private/var/tmp/_bazel_pcj/178d7438552046b1be3cba61fe7b75a8/external/bazel_tools/tools/osx/xcode_configure.bzl:91:5: Invoking xcode-locator failed, return code 256, stderr: java.io.IOException: Cannot run program "/private/var/tmp/_bazel_pcj/178d7438552046b1be3cba61fe7b75a8/external/local_config_xcode/./xcode-locator-bin" (in directory "/private/var/tmp/_bazel_pcj/178d7438552046b1be3cba61fe7b75a8/external/local_config_xcode"): error=2, No such file or directory.
</code>

Perhaps https://github.com/bazelbuild/bazel/blob/master/tools/osx/xcode_configure.bzl#L87 should check for an error (`result.return_code`) and act accordingly?  For example, on my (somewhat older 10.9.5) mac when I run the command:

``` sh
$ xcrun -v clang -fobjc-arc -framework CoreServices -framework Foundation -o xcode-locator-bin 
xcrun: note: PATH = '/Users/pcj/bin/google-cloud-sdk/bin:/Library/Java/JavaVirtualMachines/jdk1.8.0_92.jdk/Contents/Home/bin:/usr/bin:/tools:/platform-tools:/usr/local/bin:/usr/local/sbin:/usr/texbin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/opt/X11/bin:/usr/texbin'
xcrun: note: SDKROOT = '/'
xcrun: note: TOOLCHAINS = ''
xcrun: note: DEVELOPER_DIR = '/Library/Developer/CommandLineTools'
xcrun: note: XCODE_DEVELOPER_USR_PATH = ''
xcrun: note: xcrun_db = '/var/folders/cz/vr4mqxtd35v3kt8p0kct8myh0000gn/T/xcrun_db'
xcrun: note: xcrun via clang (xcrun)
xcrun: note: database key is: clang|/||/Library/Developer/CommandLineTools|
xcrun: note: lookup resolved in '/var/folders/cz/vr4mqxtd35v3kt8p0kct8myh0000gn/T/xcrun_db' : '/Library/Developer/CommandLineTools/usr/bin/clang'
Undefined symbols for architecture x86_64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by c-parsons** [comment]
Your example command is a bit incomplete in what the repository rule is running, in that you don't provide input source files. We would expect:
xcrun -v clang -fobjc-arc -framework CoreServices -framework Foundation -o xcode-locator-bin **/path/to/xcode-locator/source**
That aside, I'll investigate why it seems that this compilation is failing in the repository rule here, and provide better error messaging / behavior in the instance that it fails.

**Comment by c-parsons** [comment]
Hmm. I'm unable to reproduce this on my machine at HEAD.
Would you mind running the following and seeing what the error is ?

```
xcrun -v clang -fobjc-arc -framework CoreServices -framework Foundation -o xcode-locator-bin
/private/var/tmp/_bazel_pcj/178d7438552046b1be3cba61fe7b75a8/external/bazel_tools/tools/osx/xcode_locator.m
```

Also providing any installed versions of xcode you have would be helpful (if any)

**Comment by pcj** [comment]
You're right, that was incomplete.  Sorry about that.  

```sh
$ xcrun -v clang -fobjc-arc -framework CoreServices -framework Foundation -o xcode-locator-bin /private/var/tmp/_bazel_pcj/178d7438552046b1be3cba61fe7b75a8/external/bazel_tools/tools/osx/xcode_locator.m
xcrun: note: PATH = '/Users/pcj/bin/google-cloud-sdk/bin:/Library/Java/JavaVirtualMachines/jdk1.8.0_92.jdk/Contents/Home/bin:/usr/bin:/tools:/platform-tools:/usr/local/bin:/usr/texbin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/opt/X11/bin:/usr/texbin'
xcrun: note: SDKROOT = '/'
xcrun: note: TOOLCHAINS = ''
xcrun: note: DEVELOPER_DIR = '/Library/Developer/CommandLineTools'
xcrun: note: XCODE_DEVELOPER_USR_PATH = ''
xcrun: note: xcrun_db = '/var/folders/cz/vr4mqxtd35v3kt8p0kct8myh0000gn/T/xcrun_db'
xcrun: note: xcrun via clang (xcrun)
xcrun: note: database key is: clang|/||/Library/Developer/CommandLineTools|
xcrun: note: lookup resolved in '/var/folders/cz/vr4mqxtd35v3kt8p0kct8myh0000gn/T/xcrun_db' : '/Library/Developer/CommandLineTools/usr/bin/clang'
/private/var/tmp/_bazel_pcj/178d7438552046b1be3cba61fe7b75a8/external/bazel_tools/tools/osx/xcode_locator.m:125:40: warning: implicit declaration of function
      'LSCopyApplicationURLsForBundleIdentifier' is invalid in C99 [-Wimplicit-function-declaration]
    NSArray *array = CFBridgingRelease(LSCopyApplicationURLsForBundleIdentifier(
                                       ^
/private/var/tmp/_bazel_pcj/178d7438552046b1be3cba61fe7b75a8/external/bazel_tools/tools/osx/xcode_locator.m:125:40: warning: incompatible integer to pointer conversion passing 'int'
      to parameter of type 'CFTypeRef' (aka 'const void *') [-Wint-conversion]
    NSArray *array = CFBridgingRelease(LSCopyApplicationURLsForBundleIdentifier(
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/System/Library/Frameworks/Foundation.framework/Headers/NSObject.h:107:54: note: passing argument to parameter 'X' here
NS_INLINE id CFBridgingRelease(CFTypeRef CF_CONSUMED X) {
                                                     ^
2 warnings generated.
Undefined symbols for architecture x86_64:
  "_LSCopyApplicationURLsForBundleIdentifier", referenced from:
      _main in xcode_locator-115505.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

```sh
$ pkgutil --pkg-info=com.apple.pkg.CLTools_Executables
package-id: com.apple.pkg.CLTools_Executables
version: 6.2.0.0.1.1424975374
volume: /
location: /
install-time: 1474569535
groups: com.apple.FindSystemFiles.pkg-group com.apple.DevToolsBoth.pkg-group com.apple.DevToolsNonRelocatableShared.pkg-group
```

```sh
$ gcc -v
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 6.0 (clang-600.0.57) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin13.4.0
Thread model: posix
```

```sh
$ xcode-select -p
/Library/Developer/CommandLineTools
```

```
$ /usr/bin/xcodebuild -version
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
```

**Comment by c-parsons** [comment]
OK, seems that there is a library in use in this binary which requires MacOS SDK 10.10 or later, which isn't available in your command line tools. (This has been included or exceeded since Xcode 6)

As a shortcut, it seems reasonable to run `/usr/bin/xcodebuild -version` and terminate early if there is an error there (indicating no xcode version installed, and thus no SDKs). This should solve your issue. I'll create the fix.

**Comment by c-parsons** [comment]
This seems like a duplicate of https://github.com/bazelbuild/bazel/issues/1750, which I fixed. Does the fix mentioned in that issue not resolve the issue for you?

**Comment by pcj** [comment]
Thanks @c-parsons, this appears resolved in 0.4.0.
